packages/wallets/src/components/Base/Badge/Badge.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { Children, cloneElement, FC, ReactElement, ReactNode } from 'react';
+import classNames from 'classnames';
+import { TGenericSizes } from '../../../types';
+import { WalletText } from '../WalletText';
+import { WalletTextProps } from '../WalletText/WalletText';
+import './Badge.scss';
+
+interface BadgeComponent extends FC<BadgeProps> {
+    Icon: React.FC<BadgeIconProps>;
+    Text: React.FC<BadgeTextProps>;
+}
+interface BadgeProps {
+    children: ReactElement | ReactElement[];
+    className?: string;
+    size: Extract<TGenericSizes, 'lg' | 'md' | 'sm' | 'xs'>; // TODO: add css classes to handle the size of the badges and map it to the text sizes.
+}
+
+interface BadgeTextProps {
+    children: WalletTextProps['children'];
+    color?: WalletTextProps['color'];
+    isBold?: boolean;
+    size?: WalletTextProps['size'];
+}
+
+interface BadgeIconProps {
+    children: ReactNode;
+    className?: string;
+}
+
+const BadgeText: FC<BadgeTextProps> = ({ children, color, isBold = false, size = 'md' }) => (
+    <WalletText color={color} size={size} weight={isBold ? 'bold' : 'initial'}>
+        {children}
+    </WalletText>
+);
+
+const BadgeIcon: FC<BadgeIconProps> = ({ children, className }) => <div className={className}>{children}</div>;
+
+const Badge: BadgeComponent = ({ children, className, size = 'md' }) => (
+    <div className={classNames('wallets-badge', className)}>
+        {Children.map(children, child => cloneElement(child, { size }))}
+    </div>
+);
+
+Badge.Text = BadgeText;
```

Comment:
hmm, why making a component class a static variable within another component? is this needed? probably should be removed unless its actually needed, 

or if its sharing / reusing, then just export - e.g. export BadgeText; export BadgeIcon

Reply:
This is just an idea of what we can do with it but you are totally right we can utilize the components that we already have within `Base` and use it here instead. 🙏🏼